dynamiqs/mesolve/mesolve.py

```diff
@@ -49,27 +56,19 @@ def mesolve(
     f = compute_vmap(_mesolve, options.cartesian_batching, is_batched, out_axes)
 
     # === apply vectorized function
-    return f(H, jump_ops, psi0, tsave, exp_ops, solver, gradient, options)
+    return f(H, jump_ops, rho0, tsave, exp_ops, solver, gradient, options)
 
 
 def _mesolve(
-    H: ArrayLike | TimeArray,
-    jump_ops: list[ArrayLike | TimeArray],
-    psi0: ArrayLike,
+    H: TimeArray,
+    jump_ops: list[TimeArray],
+    rho0: ArrayLike,
     tsave: ArrayLike,
     exp_ops: list[ArrayLike] | None = None,
     solver: Solver = Tsit5(),
     gradient: Gradient | None = None,
     options: Options = Options(),
 ) -> Result:
-    # === convert arguments
-    H = _astimearray(H)
-    Ls = [_astimearray(L) for L in jump_ops]
-    y0 = jnp.asarray(psi0, dtype=cdtype())
-    y0 = todm(y0)
-    ts = jnp.asarray(tsave)
-    Es = jnp.asarray(exp_ops, dtype=cdtype()) if exp_ops is not None else None
-
```

Comment:
I 👍 keeping the arguments name in this section. However, I think we should keep the same API for `mesolve` and `_mesolve`. This makes it very handy if one wants to use the `_sesolve` method without the batching layer (`sesolve` is really made for JAX-beginners). For example, if one wants to batch over a PyTrees of `H` and not a list, which we don't support yet, it's done in two lines using `_sesolve`.

Reply:
I don't get what you want exactly. We need these argument conversions before the `vmap` to catch the object dimensions.

Reply:
Sorry I wasn't clear. We should keep the argument conversions in `mesolve`, before the `vmap`, we do need them indeed. I think we should also keep the argument conversions at the beginning of `_mesolve` (and the arguments typing). The main reason is that people who want to do custom stuff (e.g. me 😋) should be able to use `_mesolve` directly. So I'm in favor of repeating the code at the beginning of both functions.

Reply:
It's a minor comment nevermind.